spring-cloud-kubernetes-config/src/main/java/org/springframework/cloud/kubernetes/config/reload/ConfigReloadAutoConfiguration.java

```diff
@@ -111,32 +112,34 @@ public ConfigurationChangeDetector propertyChangeWatcher(
 		@Bean
 		@ConditionalOnMissingBean
 		public ConfigurationUpdateStrategy configurationUpdateStrategy(
-				ConfigReloadProperties properties, ConfigurableApplicationContext ctx,
-				RestartEndpoint restarter, ContextRefresher refresher) {
+			ConfigReloadProperties properties, ConfigurableApplicationContext ctx,
+			@Autowired(required = false) RestartEndpoint restarter,
+			ContextRefresher refresher) {
 			switch (properties.getStrategy()) {
 			case RESTART_CONTEXT:
+				Assert.notNull(restarter, "Restart endpoint is not enabled");
```

Comment:
this is the main change right?  it is hard to see with the formatting changes

Reply:
one change, an autowired injected param noted below that made optional.